### PR TITLE
Scripts for "New vehicle data GearStatus" feature

### PR DESCRIPTION
**Pull request by mkorniichuk**
ATF Test Scripts to check [#3216 New vehicle data GearStatus](https://github.com/smartdevicelink/sdl_core/issues/3216)

This PR is **ready** for review.

### Summary
Scripts for "New vehicle data GearStatus" feature

### ATF version
[develop](https://github.com/LuxoftSDL/sdl_atf/commit/45fa34d82a2a6fdfa5874dfeb5b94f25c4ade446)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/API/VehicleData/GearStatus/007_SubscribeVD_success_flow_with_gearStatus_param.lua [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---------------------------------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (GetmanetsIrina) [reply test_scripts/API/VehicleData/GearStatus/007_SubscribeVD_success_flow_with_gearStatus_param.lua:1]: Scripts with resultCode IGNORED are added in 3496e1e

**Comment by atiwari9** [comment]
Test Report for commit: [b4e023d](https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2417/commits/b4e023d4204dc85a3b2889e6fd525512190a8837)

Logs: 

[Test_Report_GearStatus_b4e023d.zip](https://github.com/smartdevicelink/sdl_atf_test_scripts/files/4786424/Test_Report_GearStatus_b4e023d.zip)


```
=====================================================================================================
TOTAL: 35
PASSED: 26 
FAILED: 0 
ABORTED: 0 
SKIPPED: 9 
03 ;./test_scripts/API/VehicleData/GearStatus/003_GetVehicleData_with_resultCode_GENERIC_ERROR_in_case_HMI_responds_with_invalid_data.lua https://github.com/smartdevicelink/sdl_core/issues/3368
12 ;./test_scripts/API/VehicleData/GearStatus/012_SubscribeVD_not_success_codes_for_gearStatus_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
19 ;./test_scripts/API/VehicleData/GearStatus/019_UnsubscribeVD_not_success_codes_for_gearStatus_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
23 ;./test_scripts/API/VehicleData/GearStatus/023_OnVehicleData_Disallowed_by_policies_Ignored_flow.lua https://github.com/smartdevicelink/sdl_core/issues/3365
24 ;./test_scripts/API/VehicleData/GearStatus/024_OnVehicleData_Disallowed_after_PTU.lua https://github.com/smartdevicelink/sdl_core/issues/3365
25 ;./test_scripts/API/VehicleData/GearStatus/025_OnVehicleData_notification_with_invalid_data.lua https://github.com/smartdevicelink/sdl_core/issues/3368
29 ;./test_scripts/API/VehicleData/GearStatus/029_Processing_deprecated_prndl_parameter.lua https://github.com/smartdevicelink/sdl_core/issues/3309
32 ;./test_scripts/API/VehicleData/GearStatus/032_Resumption_of_subscription_for_2_apps_to_the_gearStatus_param_after_an_Unexpected_Disconnect.lua https://github.com/smartdevicelink/sdl_core/issues/2283
33 ;./test_scripts/API/VehicleData/GearStatus/033_Resumption_of_subscription_for_2_apps_to_the_gearStatus_param_after_IGN_OFF_ON.lua https://github.com/smartdevicelink/sdl_core/issues/2283
=====================================================================================================
```

**Comment by LitvinenkoIra** [comment]
@atiwari9 please review this PR

**Comment by atiwari9** [comment]
Another test report for commit:[b4e023d](https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2417/commits/b4e023d4204dc85a3b2889e6fd525512190a8837)

```
=====================================================================================================
TOTAL: 35
PASSED: 26 
FAILED: 0 
ABORTED: 0 
SKIPPED: 9 
03 ;./test_scripts/API/VehicleData/GearStatus/003_GetVehicleData_with_resultCode_GENERIC_ERROR_in_case_HMI_responds_with_invalid_data.lua https://github.com/smartdevicelink/sdl_core/issues/3368
12 ;./test_scripts/API/VehicleData/GearStatus/012_SubscribeVD_not_success_codes_for_gearStatus_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
19 ;./test_scripts/API/VehicleData/GearStatus/019_UnsubscribeVD_not_success_codes_for_gearStatus_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
23 ;./test_scripts/API/VehicleData/GearStatus/023_OnVehicleData_Disallowed_by_policies_Ignored_flow.lua https://github.com/smartdevicelink/sdl_core/issues/3365
24 ;./test_scripts/API/VehicleData/GearStatus/024_OnVehicleData_Disallowed_after_PTU.lua https://github.com/smartdevicelink/sdl_core/issues/3365
25 ;./test_scripts/API/VehicleData/GearStatus/025_OnVehicleData_notification_with_invalid_data.lua https://github.com/smartdevicelink/sdl_core/issues/3368
29 ;./test_scripts/API/VehicleData/GearStatus/029_Processing_deprecated_prndl_parameter.lua https://github.com/smartdevicelink/sdl_core/issues/3309
32 ;./test_scripts/API/VehicleData/GearStatus/032_Resumption_of_subscription_for_2_apps_to_the_gearStatus_param_after_an_Unexpected_Disconnect.lua https://github.com/smartdevicelink/sdl_core/issues/2283
33 ;./test_scripts/API/VehicleData/GearStatus/033_Resumption_of_subscription_for_2_apps_to_the_gearStatus_param_after_IGN_OFF_ON.lua https://github.com/smartdevicelink/sdl_core/issues/2283
=====================================================================================================
```

Logs:[Test_Report_Gear_Status_b4e023d_2.zip](https://github.com/smartdevicelink/sdl_atf_test_scripts/files/4793880/Test_Report_Gear_Status_b4e023d_2.zip)

**Comment by atiwari9** [comment]
PR Approved.

**Comment by LitvinenkoIra** [comment]
@theresalech PR is ready for Livio review. Thank you!

**Comment by ShobhitAd** [comment]
@LitvinenkoIra In reference to https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2421#issuecomment-657482958. I think this vehicle data PR should also be updated based on https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2413#discussion_r451132043.

**Comment by ShobhitAd** [comment]
@LitvinenkoIra just as a note, in reference to https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2413#issuecomment-664457788, the updates to this PR should be based off of the `feature/vehicle_data_testing` branch.

**Comment by dboltovskyi** [comment]
@ShobhitAd Please notice as was agreed previously current PR has been completely reworked based on `feature/vehicle_data_testing` branch and is ready for PM review.

**Pull request merged**